### PR TITLE
Move the GPG Key and SSL Certificate details to its own page

### PR DESCRIPTION
**Pull request by danidoni**
The GPG Key and SSL Certificate details was present in a modal embedded into
the Project overview. On small devices it was harder to see all the data as you
needed to scroll to the right.

Extracting the details into a new page provides more space. We also decided not
to use a table to present the data to make it easier to select and copy the
interesting bits.

Here is a screenshot of how it looks:

**Before**
Screenshot_20200228_115519

**After**
_Extra small screens_
Screenshot_20200302_144654

_Small screens_
Screenshot_20200302_144751

_Medium screens_
Screenshot_20200302_144811

_Large screens and above_
Screenshot_20200302_145442

### CODE REVIEW

File: src/api/app/assets/stylesheets/webui/responsive_ux/_keys_and_certs.scss [hunk 1]
```diff
@@ -0,0 +1,92 @@
+.key-and-certificates-card {
```
Comment (hennevogel) [inline src/api/app/assets/stylesheets/webui/responsive_ux/_keys_and_certs.scss]: Hm, I don't understand. What is the use to just "rename" the bootstrap classes?

File: src/api/app/views/webui/project/keys_and_certificates.html.haml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+              = link_to(project_public_key_path(project_name: @project.name)) do
+                %i
+                GPG Key
+      .ssl_cert_column
+        .card
+          .card-body
+            %h5.card-title SSL Certificate
+            %dl
+              %dt Fingerprint
+              %dd= @project.key_info.fingerprint
```
Comment (hennevogel) [inline src/api/app/views/webui/project/keys_and_certificates.html.haml]: This is the fingerprint of the GPG key, it has nothing to do with the SSL certificate.

**Comment by vpereira** [comment]
I still prefer as GH does:

Screenshot_2020-02-28_09-27-21

**Comment by hennevogel** [comment]
I don't like the design too much, sorry. This is too basic. Have you thought about what is most important about the key? Here are some inspirations:

## Github
Screenshot_2020-02-28_09-27-21
## Gitlab
Screenshot from 2020-02-28 14-20-33
## Circle-ci
Screenshot from 2020-02-28 14-22-56

**Comment by vpereira** [comment]
I still prefer how GH display it.. instead of the delete, the download button:
 
Screenshot_2020-02-28_14-22-14

File: src/api/app/views/webui/project/keys_and_certificates.html.haml (hunk 2 above)
Comment (danidoni) [reply src/api/app/views/webui/project/keys_and_certificates.html.haml]: True, fixing

File: src/api/app/assets/stylesheets/webui/responsive_ux/_keys_and_certs.scss (hunk 1 above)
Comment (danidoni) [reply src/api/app/assets/stylesheets/webui/responsive_ux/_keys_and_certs.scss]: In this case its not super-clear what's the benefit of doing it so, but if you look at [this other element](https://github.com/openSUSE/open-build-service/pull/9156/files#diff-e97de64bb90a9262713f71a9c2538b84R17), placing the bootstrap classes inline would render this:
```
.row
  .col-12.col-md-6.mb-2.p-3.p-md-3
    .card
```
Which is uglier...
And there's also the benefit of using some classes with meaningful semantics

Comment (danidoni) [reply src/api/app/assets/stylesheets/webui/responsive_ux/_keys_and_certs.scss]: Anyway, @saraycp also showed concerns regarding this use of `@extend` so I will fall back to my previous changes that were inlined

**Comment by danidoni** [comment]
> I don't like the design too much, sorry. This is too basic. Have you thought about what is most important about the key? Here are some inspirations:
> ## Github
> 
> Screenshot_2020-02-28_09-27-21
> ## Gitlab
> 
> Screenshot from 2020-02-28 14-20-33
> ## Circle-ci
> 
> Screenshot from 2020-02-28 14-22-56

Okay, I'll look at them and come up with something

File: src/api/app/assets/stylesheets/webui/responsive_ux/_keys_and_certs.scss (hunk 1 above)
Comment (hennevogel) [reply src/api/app/assets/stylesheets/webui/responsive_ux/_keys_and_certs.scss]: > Which is uglier...

Uglier how? If I know bootstrap, I perfectly understand what is going on there. And hiding this in some CSS file removes my understanding of what is going on.

Comment (danidoni) [reply src/api/app/assets/stylesheets/webui/responsive_ux/_keys_and_certs.scss]: Its uglier in the sense that it has no semantic, this line could be anywhere else and no one cared.

About hiding the classes in some CSS I can only say that I used the same logic I use to abstract some code behind a function or a class. So `.key-and-certificates-card` is useful to me in telling me what this is right away, but `.col-12.col-md-6.mb-2.p-3.p-md-3` at first glance is just gibberish... Don't you think?

As I said earlier, I'm falling back to inline the bootstrap classes for consistency's sake.

File: src/api/app/views/webui/project/keys_and_certificates.html.haml (hunk 2 above)
Comment (danidoni) [reply src/api/app/views/webui/project/keys_and_certificates.html.haml]: Fixed

**Comment by danidoni** [comment]
> Please squash the commits together

Done

File: src/api/app/views/webui/project/keys_and_certificates.html.haml [hunk 3]
```diff
@@ -0,0 +1,56 @@
+- @pagetitle = "Key and certificate for #{@project}"
+
+.card.mb-3
+  = render partial: 'tabs', locals: { project: @project }
```
Comment (hennevogel) [inline src/api/app/views/webui/project/keys_and_certificates.html.haml]: You render tabs but have no entry in it. Is this intentional?

Comment (danidoni) [reply src/api/app/views/webui/project/keys_and_certificates.html.haml]: Yes, I want to signal that we're still in the "same" place than before, but we have no entry in the tab.

Comment (hennevogel) [reply src/api/app/views/webui/project/keys_and_certificates.html.haml]: Then this should make `Overview` active right?

Comment (danidoni) [reply src/api/app/views/webui/project/keys_and_certificates.html.haml]: Right

Comment (danidoni) [reply src/api/app/views/webui/project/keys_and_certificates.html.haml]: Done

**Pull request merged**